File: spigot/src/main/java/kr/entree/samplespigot/SpigradleSample.java
package kr.entree.samplespigot;

import org.bukkit.plugin.PluginDescriptionFile;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.plugin.java.JavaPluginLoader;

import java.io.File;

public class SpigradleSample extends JavaPlugin {
    public SpigradleSample() {
    }

    // Constructor for MockBukkit
    public SpigradleSample(JavaPluginLoader loader, PluginDescriptionFile description, File dataFolder, File file) {
        super(loader, description, dataFolder, file);
    }

    @Override
    public void onEnable() {
        getLogger().info("This is Spigot plugin!");
    }
}

File: spigot-kotlin/src/main/kotlin/kr/entree/samplespigot/SpigradleSample.kt
package kr.entree.samplespigot

import org.bukkit.plugin.PluginDescriptionFile
import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.plugin.java.JavaPluginLoader
import java.io.File

class SpigradleSample : JavaPlugin {
    constructor() : super()

    // Constructor for MockBukkit. if you don't need it, use: "class SpigradleSample() : JavaPlugin()" without both constructors.
    constructor(loader: JavaPluginLoader, description: PluginDescriptionFile, dataFolder: File, file: File?) : super(loader, description, dataFolder, file
            ?: File(""))

    override fun onEnable() {
        logger.info("This is Spigot plugin!");
    }
}